### PR TITLE
Updated resources.json

### PR DESCRIPTION
**Pull request by naever32**
Added a learning resource.

### CODE REVIEW

File: static/resources.json [hunk 1]
```diff
@@ -347,6 +347,18 @@
           }
         ]
       },
+      "Python Principles": {
+        "description": "An online Python course for total beginners. Lets you learn through clear explanations and lots of practical exercises. Very hands-on.",
+        "payment": "optional",
```
Comment (sco1) [inline static/resources.json:352]:
```suggestion
        "payment": "paid",
```

While the free tier provides a glimpse into the service & its offerings, realistically this is a paid service if you want to end up with an understanding of Python.

**Review by sco1 — CHANGES_REQUESTED** [review]
Thanks for the link! From what's visible for free this seems to be a very well put together site. We'll have some more folks check it out in depth before a decision to merge is made.

For future reference, please include a more descriptive title & description for the PR so reviewers have some idea what they're looking at. Same goes for commit messages, as explained in our [Contributors Guideline](https://github.com/python-discord/site/blob/master/CONTRIBUTING.md).

**Comment by sco1** [comment]
The site appears to be running Python 2, which means it does not support a whole host of invaluable modern Python features. With [Python 2 EOL coming 2020-01-01](https://python3statement.org/), I can't recommend adding this to our resources until it's been updated.

image

**Comment by naever32** [comment]
> For future reference, please include a more descriptive title & description for the PR so reviewers have some idea what they're looking at. Same goes for commit messages, as explained in our [Contributors Guideline](https://github.com/python-discord/site/blob/master/CONTRIBUTING.md).

I will keep this in mind for next time, sorry. Did not know it was py2.

**Comment by lemonsaurus** [comment]
@naever32 that's alright. FYI we've been in touch with the creator and they say they're working on porting this over to Python 3. We will probably be open to adding this resource once that's been done, so we'll stay in touch with them and reopen this ticket once that happens. 

I'm gonna close it for now, though.

**Comment by tehybel** [comment]
Hi guys,
The site has been running Python 3 for a while now:
version
@sco1 are you willing to reconsider adding it to the learning resources list? 

(Just let me know if any of you need a free membership to test out all the lessons. I run the site.)

By the way, the site has a bunch of free challenges now (https://pythonprinciples.com/challenges/), so perhaps setting `"payment": "optional"` may make sense at this point, though that's up to you.

**Comment by sco1** [comment]
Hello, welcome back!

I’ll make a note for folks to check it out again and we’ll go from there!

**Comment by lemonsaurus** [comment]
@tehybel I'm giving it a spin now, and running into a strange bug with some of the free content. Any idea why this perfectly valid 3.6 feature is being interpreted as a SyntaxError?

image

**Comment by lemonsaurus** [comment]
@tehybel here's another interesting thing that should've worked in 3.7:

image

**Comment by lemonsaurus** [comment]
This feature is 3.5 
image

**Comment by lemonsaurus** [comment]
meanwhile, this feature works and it's 3.7 only: 
image

Looks like you're running some sort of pseudo-3.7

**Comment by lemonsaurus** [comment]
It doesn't even support f-strings, which are sort of the keystone python 3 feature that everyone loves - 
image

**Comment by lemonsaurus** [comment]
@tehybel 

Overall, I think this project shows promise and there are definitely good parts, but I don't think we can recommend this in our resources list until it supports (and teaches) modern Python 3 features, like f-strings, type annotations and asyncio.

Feel free to let us know if you add support for these things, and we'll be glad to reconsider it.

**Comment by tehybel** [comment]
@lemonsaurus Thank you for taking the time to tell me about all this, it really is appreciated. There is definitely a bug causing most of these issues; I think the module that parses the syntax tree of user input to provide helpful feedback needs an update. I'll look into it.

I don't believe that type annotation and asyncio are in scope for this course, since it is for complete beginners. Expanding the course contents to cover more advanced concepts isn't a priority at this time.

Either way, thanks again for your time, and I'll look into the issues you described.

**Comment by lemonsaurus** [comment]
@tehybel

I suppose asyncio might fall outside the scope of a beginners course, but I think a precursory glance at how type annotations look and work would fit well in a beginners course. it is, after all, syntax. it's not very difficult, and I don't think a beginners course should leave its takers confused about basic syntax when they encounter it.

**Pull request closed without merging**